neo/Prompt/Commands/Config.py

```diff
@@ -0,0 +1,205 @@
+            elif c1 == 'off' or c1 == '0':
+                settings.set_log_smart_contract_events(False)
+                print("Smart contract event logging is now disabled")
+                return c1
+            else:
+                print("Cannot configure log. Please specify on|off")
+                return
+        else:
+            print("Cannot configure log. Please specify on|off")
+            return
+
+    def command_desc(self):
+        p1 = ParameterDesc('attribute', 'either "on"|"off" or 1|0')
+        return CommandDesc('sc-events', 'toggle smart contract events on or off', [p1])
+
+
+class CommandConfigDebugNotify(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        c1 = get_arg(arguments)
+        if c1 is not None:
+            c1 = c1.lower()
+            if c1 == 'on' or c1 == '1':
+                settings.set_emit_notify_events_on_sc_execution_error(True)
+                print("Smart contract emit Notify events on execution failure is now enabled")
+                return c1
+            elif c1 == 'off' or c1 == '0':
+                settings.set_emit_notify_events_on_sc_execution_error(False)
+                print("Smart contract emit Notify events on execution failure is now disabled")
+                return c1
+            else:
+                print("Cannot configure log. Please specify on|off")
+                return
+        else:
+            print("Cannot configure log. Please specify on|off")
+            return
+
+    def command_desc(self):
+        p1 = ParameterDesc('attribute', 'either "on"|"off" or 1|0')
+        return CommandDesc('sc-debug-notify', 'toggle Notify events on execution failure on or off', [p1])
+
+
+class CommandConfigVMLog(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        c1 = get_arg(arguments)
+        if c1 is not None:
+            c1 = c1.lower()
+            if c1 == 'on' or c1 == '1':
+                settings.set_log_vm_instruction(True)
+                print("VM instruction execution logging is now enabled")
+                return c1
+            elif c1 == 'off' or c1 == '0':
+                settings.set_log_vm_instruction(False)
+                print("VM instruction execution logging is now disabled")
+                return c1
+            else:
+                print("Cannot configure VM instruction logging. Please specify on|off")
+                return
+        else:
+            print("Cannot configure VM instruction logging. Please specify on|off")
+            return
+
+    def command_desc(self):
+        p1 = ParameterDesc('attribute', 'either "on"|"off" or 1|0')
+        return CommandDesc('vm-log', 'toggle VM instruction execution logging on or off', [p1])
+
+
+class CommandConfigNodeRequests(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        if len(arguments) in [1, 2]:
```

Comment:
Right now we can have the following output
```
neo> config node-requests -1 bla
Please specify a block request part and max size for each node, like 30 and 1000
neo> config node-requests -1 -1
[I 181219 09:30:07 NodeLeader:146] invalid values. Please specify a block request part and max size for each node, like 30 and 1000
```

I'd like to see uniform error messages

To solve this you can change
https://github.com/CityOfZion/neo-python/blob/e3780126f9cf8175f9f3c7ad27f36f2baae20508/neo/Network/NodeLeader.py#L143
to raise a `ValueError` instead of printing a message, so you can act on it (and other interfaces maybe in the future).

Reply:
I kept the original message, in case this function is used separately.